PukingPredator/Assets/Scripts/GoalStates/ObjectiveTracker.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ObjectiveTracker : MonoBehaviour
+{
+    /*
+     The player must collect all objectives to win
+
+    Objectives get the "Objective" script, and this script will register all objectives upon loading.
+    The objectives will call functions in this objectiveTracker when they are collected.
+     */
+    private GameObject objectsLeftObject;
+    private Text objectsLeftText;
+
+    private int remainingCollectionObjects;
+
+    void Start()
+    {
+        objectsLeftObject = GameObject.Find("objectsLeftText");
+        objectsLeftText = objectsLeftObject.GetComponent<Text>();
+
+        // find all objectives
+        Objective[] objectives = FindObjectsOfType<Objective>();

```

Comment:
For now its not too much of an issue but if theres many objects in the level this might take a while cause it looks through all items. This can be a fix for later but simply attaching the component to the Objective types and then firing off an event if the player collides could be faster

Reply:
oh I see, yeah we could do that later if this becomes an issue